app/api/mailer/route.ts

```diff
@@ -0,0 +1,60 @@
+import { NextApiResponse } from 'next';
+import { NextResponse } from 'next/server';
+import nodemailer from 'nodemailer';
+
+const email = process.env.EMAIL_SERVER_USER;
+const pass = process.env.EMAIL_SERVER_PASSWORD;
+const port = process.env.EMAIL_SERVER_PORT;
+const host = process.env?.EMAIL_SERVER_HOST ?? 'smtp.gmail.com';
+
+export const transporter = nodemailer.createTransport({
+  host,
+  port,
+  secure: true,
+  service: 'gmail',
+  auth: {
+    type: 'login',
+    user: email,
+    pass,
+  },
+});
+
+const CONTACT_MESSAGE_FIELDS = {
+  name: 'Name',
+  email: 'Email',
+  subject: 'Subject',
+  message: 'Message',
+};
+
+export async function POST(req: Request, res: NextApiResponse) {
```

Comment:
Trong app router thì mình không dùng `NextApiResponse`, nên dùng;

```
export async function GET/POST(req: NextRequest) {

    return NextResponse.json(result);
}
```
https://nextjs.org/docs/app/building-your-application/routing/route-handlers#extended-nextrequest-and-nextresponse-apis